chips/lowrisc/src/usbdev.rs

```diff
@@ -422,6 +463,199 @@ impl<'a> Usb<'a> {
                 break;
             }
         }
+
+        self.bufs.set(bufs);
+    }
+
+    fn stall(&self, endpoint: usize) {
+        self.registers
+            .stall
+            .set(1 << endpoint | self.registers.stall.get());
+    }
+
+    fn copy_slice_out_to_hw(&self, ep: usize, buf_id: usize, size: usize) {
```

Comment:
This should use enumerations rather than usize, for ep and buf_id, to allow compile-time rather than runtime checking.

Reply:
That would required converting from the uszie from the HIL to a enum, what advantage does that give us? It is then written to the hardware as just a number.